mtg_card_identifier/CardScanner/CardScannerFile.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  CardScannerFile.swift
+//  mtg_card_identifier
+//
+//  Created by mac on 8/16/23.
+//
+
+import SwiftUI
+import UIKit
+import VisionKit
+
+struct LiveTextScanner: UIViewControllerRepresentable {
+    
+    // Will need more than just a string when we are doing the api call.
+    var scanedText: Binding<[String]>
+    var style: Style
+    
+    
+    
+    func makeCoordinator() -> Coordinator {
+        Coordinator(scanedTextBinding: scanedText)
+    }
+    
+    func makeUIViewController(context: Context) -> DataScannerViewController {
+        
+//        The x and y needs the math part, going from the window to access the screen which will let you work with the bounds.
+        
+        
+        let vc = DataScannerViewController(
+            recognizedDataTypes: [.text()],
+            qualityLevel: .balanced,
+            recognizesMultipleItems: true,
+            isHighFrameRateTrackingEnabled: true
+            //            isGuidanceEnabled: true,
+            //            Don't know if we want that to work but it's basically done
+            //            isHighlightingEnabled: true
+        )
+        
+        
+        let partialTransparentView = PartialTransparentView(cutout: style.overlay)
+        partialTransparentView.translatesAutoresizingMaskIntoConstraints = true
+        partialTransparentView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        partialTransparentView.frame = vc.overlayContainerView.bounds
+        
+        vc.view.addSubview(partialTransparentView)
+        vc.regionOfInterest = style.regionOfInterest
+        vc.delegate = context.coordinator
+        
+        do {
+            try vc.startScanning()
+            
+        } catch {
+            
+            print("Error: \(error.localizedDescription)")
```

Comment:
What kinds of errors can we get here? Does the documentation say anything?

Reply:
I wonder if they have disallowed the authorization if that causes them to end up here. If so, maybe we could get rid of the if/else CardScannnerView and have the “else” view (the one that says you need to enable permissions with a link to the Settings app) returned here. 
In order to do that, you’d have to change the generic value of this object to be UIView rather than DataScannerViewController (essentially update the function signature of makeViewController and updateViewController.

Reply:
I’d probably have this be a variable on the LiveTextScanner object and have the person that initializes it provide the SwiftUI view to embed. Then this would return `UIHostingController(rootView: noPermissionsEnabledViewThatIsPassedIntoThisView)`.